slack_sdk/http_retry/builtin_handlers.py

```diff
@@ -0,0 +1,101 @@
+import random
+import time
+from http.client import RemoteDisconnected
+from typing import Optional, List
+from urllib.error import URLError
+
+from slack_sdk.http_retry.interval_calculator import RetryIntervalCalculator
+from slack_sdk.http_retry.state import RetryState
+from slack_sdk.http_retry.request import HttpRequest
+from slack_sdk.http_retry.response import HttpResponse
+from slack_sdk.http_retry.handler import RetryHandler, default_interval_calculator
+
+
+class ConnectionErrorRetryHandler(RetryHandler):
+    """RetryHandler that does retries for connectivity issues."""
+
+    def __init__(
+        self,
+        max_retry_count: int = 1,
+        interval_calculator: RetryIntervalCalculator = default_interval_calculator,
+        error_types: List[Exception] = [
+            # To cover URLError: <urlopen error [Errno 104] Connection reset by peer>
+            URLError,
+            ConnectionResetError,
+            RemoteDisconnected,
+        ],
+    ):
+        super().__init__(max_retry_count, interval_calculator)
+        self.error_types_to_do_retries = error_types
+
+    def _can_retry(
+        self,
+        *,
+        state: RetryState,
+        request: HttpRequest,
+        response: Optional[HttpResponse],
+        error: Optional[Exception],
+    ) -> bool:
+        if error is None:
+            return False
+
+        for error_type in self.error_types_to_do_retries:
+            if isinstance(error, error_type):
+                return True
+        return False
+
+
+class ServerErrorRetryHandler(RetryHandler):
```

Comment:
I've added this one as a reference implementation but it's unused. We may want to remove this for now.

Reply:
Yes, this is fair enough 👍

Reply:
Fair!

Reply:
This feels like a good time to mention the [Circuit Breaker pattern](https://learn.microsoft.com/en-us/azure/architecture/patterns/circuit-breaker).